### PR TITLE
fix(core): Announce new webviews and windows

### LINKED ISSUE #8144: [bug] v2 - Dynamically created windows are not being updated at 'getAll' method.
## Description

On JS api, child windows created with `new Window()` are not being returned in the `getAll` of the parent window.

## Details

I have a list of all the current opened windows in my application and a button "Create new Window" that opens up a new one. If I click the button I want my list to be mapped with the `getAll` returned updated values (both parent and childs).

## How to

- Create a new window with the js API of tauri-plugin-window:

```ts
import { Window, getAll } from '@tauri-apps/plugin-window';

// all inside an async function
const webview = new Window('window_1', {
    url: '/',
    resizable: true,
    focus: true,
    title: `Test`,
  });
await webview.once('tauri://created', () => {
  console.info('New window created.');
  console.log(getAll()) // show parent and child windows
  return webview;
});
await webview.once('tauri://error', (e: any) => {
  console.info(e);
});

console.log(getAll()) // show only parent window.
```

If I log `getAll()` inside the `webview.once('tauri://created', () => {})` event, it will return me both parent and child windows. However, if I log `getAll()` anywhere outside the event, it will return only the parent window.

Is there some async thing that I'm missing?

### Versions

```
@tauri-apps/cli: 2.0.0-alpha.14
@tauri-apps/api: 2.0.0-alpha.8
@tauri-apps/plugin-clipboard-manager: 2.0.0-alpha.1
@tauri-apps/plugin-window: 2.0.0-alpha.1
```

### LINKED ISSUE #9200: [bug][v2] Failed to get dynamically created window
### Describe the bug

The window created through the 'WebviewWindow' of the JS API cannot be obtained.

### Reproduction

```vue
<!-- App.vue -->
<script setup lang="ts">
import { WebviewWindow, getCurrent } from '@tauri-apps/api/webviewWindow'

const label = getCurrent().label

async function createWindow(
): Promise<WebviewWindow> {
  const window = WebviewWindow.getByLabel('window')
  console.log('🔍️ > window:', window)
  if (window) {
    return window
  }
  const windows = WebviewWindow.getAll()
  console.log('🔍️ > windows:', windows)
  return new WebviewWindow('window', {
    title: 'window',
    url: '/',
  })
}
</script>

<template>
  <div class="container">
    <div style="margin-bottom: 2rem;font-size: 3rem;font-weight: 600;">{{ label }}</div>
    <button @click="createWindow()">
      Open Window
    </button>
  </div>
</template>
```

Click the Open Window button to open the preview window.  Click it again, and it will attempt to create another window and report an error in the console.

### Expected behavior

The `WebviewWindow.getByLabel` method should return a `WebviewWindow` object.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22631 X64
    ✔ WebView2: 122.0.2365.80
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.76.0 (07dca489a 2024-02-04)
    ✔ cargo: 1.76.0 (c84b36747 2024-01-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.10.0
    - pnpm: 8.15.4
    - yarn: 3.6.1
    - npm: 10.2.3

[-] Packages
    - tauri [RUST]: 2.0.0-beta.11
    - tauri-build [RUST]: 2.0.0-beta.9
    - wry [RUST]: 0.37.0
    - tao [RUST]: 0.26.1
    - @tauri-apps/api [NPM]: 2.0.0-beta.5
    - @tauri-apps/cli [NPM]: 2.0.0-beta.9

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by FabianLars**
fixes https://github.com/tauri-apps/tauri/issues/9200
fixes https://github.com/tauri-apps/tauri/issues/8144

### CODE REVIEW

File: core/tauri/src/manager/webview.rs [hunk 1]
```diff
@@ -610,6 +610,19 @@ impl<R: Runtime> WebviewManager<R> {
...
+      "window.__TAURI_INTERNALS__.metadata.webviews = {webview_labels_array}.map(function (label) {{ return {{ label: label }} }})",
+    ));
+    }
+
+    let _ = webview.manager.emit(
+      "tauri://webview-created",
+      Some(crate::webview::CreatedEvent {
+        label: webview.label().into(),
+      }),
+    );
```
Comment (amrbashir) [inline core/tauri/src/manager/webview.rs:624]: why emit the same events here as the one in `WebviewBuilder::build`?

Comment (FabianLars) [reply core/tauri/src/manager/webview.rs:624]: ahh i may not have tested multi webview windows, can't remember. We can probably remove the event in WebviewBuilder::build ~~(or move this one here into the if let some in WindowBuilder::build_internal)~~

Comment (amrbashir) [reply core/tauri/src/manager/webview.rs:624]: let's just remove this one and keep the one in `WebviewBuilder::build`

Comment (FabianLars) [reply core/tauri/src/manager/webview.rs:624]: The one in WebViewBuilder doesn't work for single webview windows. That's why I added this one here in the first place. Not sure if it's worth it right now to look into reusing the WebViewBuilder for single webview windows while it's unstable but if you prefer that I can try to do that.

Comment (amrbashir) [reply core/tauri/src/manager/webview.rs:624]: I see, then just remove the one in WebviewBuilder and keep this one instead

**Review by amrbashir — APPROVED** [review]

**Pull request merged**